src/main/java/seedu/address/model/task/TaskName.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.model.task;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Task's name in the todo list.
+ * Guarantees: immutable; is valid as declared in {@link #isValidTaskName(String)}
+ */
+public class TaskName {
+    public static final String MESSAGE_CONSTRAINTS = "Task name should be no longer than 30 words.";
+
+    private final String value;
+
+    /**
+     * Constructs a {@code TaskName}.
+     *
+     * @param taskName valid task name.
+     */
+    public TaskName(String taskName) {
+        requireNonNull(taskName);
+        this.value = taskName;
+    }
+
+    /**
+     * Returns true if a given string is a valid task name.
+     *
+     * @param test given string
+     * @return true if task name is valid
+     */
+    public static boolean isValidTaskName(String test) {
+        return test.length() <= 30;
```

Comment:
Don't use magic numbers, declare 30 as a constant in the field, and name it the word limit or something. This makes it more readable.

Reply:
Okay, noted.